# 📘 Pre-entrega

**Nombre del alumno:** Daniela Perea

**Nombre del alumno:** 25262

## 🧩 Etapa 1: Recopilación y Preparación de Datos
**Objetivo:** Demostrar habilidades en Python, familiaridad con el entorno de trabajo y conocimientos básicos sobre manipulación de datos.

### 1️⃣ Carga de datos

In [1]:
# Importar Librerías
import pandas as pd
import numpy as np
import os

In [2]:
# Cargar los sets de datos en DataFrames

## Montar la unidad
from google.colab import drive
drive.mount('/content/drive')

## Verificar que los archivos csv se encuentren en la carpeta datasets
os.listdir('/content/drive/MyDrive/Data Analytics con Python/Perea Daniela - Comisión 25262 - TPI Data Analytics - Pre-entrega/Datasets/')

## Definir las rutas de los datasets.
ruta_ventas = "/content/drive/MyDrive/Data Analytics con Python/Perea Daniela - Comisión 25262 - TPI Data Analytics - Pre-entrega/Datasets/ventas.csv"
ruta_clientes = "/content/drive/MyDrive/Data Analytics con Python/Perea Daniela - Comisión 25262 - TPI Data Analytics - Pre-entrega/Datasets/clientes.csv"
ruta_marketing = "/content/drive/MyDrive/Data Analytics con Python/Perea Daniela - Comisión 25262 - TPI Data Analytics - Pre-entrega/Datasets/marketing.csv"

## Cargar CSV como DataFrames.

ventas = pd.read_csv(ruta_ventas)
clientes = pd.read_csv(ruta_clientes)
marketing = pd.read_csv(ruta_marketing)



Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [3]:
# Verificar Datasets

## Validar formas para comprobar que se cargaron correctamente.
print("ventas.shape ->", ventas.shape)
print("clientes.shape ->", clientes.shape)
print("marketing.shape ->", marketing.shape)

## Mostrar primeras filas
display(ventas.head(3))
print("\n")
display(clientes.head(3))
print("\n")
display(marketing.head(3))

ventas.shape -> (3035, 6)
clientes.shape -> (567, 5)
marketing.shape -> (90, 6)


id_venta           producto   precio  cantidad fecha_venta  \
0       792  Cuadro decorativo   $69.94       5.0  02/01/2024   
1       811    Lámpara de mesa  $105.10       5.0  02/01/2024   
2      1156           Secadora   $97.96       3.0  02/01/2024   

           categoria  
0         Decoración  
1         Decoración  
2  Electrodomésticos

id_cliente               nombre  edad         ciudad  ingresos
0           1      Aloysia Screase    44  Mar del Plata  42294.68
1           2  Kristina Scaplehorn    25        Posadas  24735.04
2           3       Filip Castagne    50    Resistencia  35744.85

id_campanha         producto  canal  costo fecha_inicio   fecha_fin
0           74  Adorno de pared     TV   4.81   20/03/2024  03/05/2024
1           12           Tablet   RRSS   3.40   26/03/2024  13/05/2024
2           32  Lámpara de mesa  Email   5.54   28/03/2024  20/04/2024

In [4]:
ventas.dtypes

id_venta         int64
producto        object
precio          object
cantidad       float64
fecha_venta     object
categoria       object
dtype: object

### 4️⃣ Análisis Exploratorio Inicial

In [5]:
def eda(df, nombre):
    print(f"=== {nombre} ===")
    print("shape:", df.shape)
    print("columnas:", list(df.columns))
    print("dtypes:")
    print(df.dtypes)
    print("\nNulos por columna:")
    print(df.isna().sum())
    print("\nPrimeras filas:")
    display(df.head(5))
    print("\nAnálisis Descriptivo (numérico):")
    display(df.describe(include='number'))
    print("-"*100)

In [6]:
eda(ventas, "VENTAS (Análisis Exploratorio)")

=== VENTAS (Análisis Exploratorio) ===
shape: (3035, 6)
columnas: ['id_venta', 'producto', 'precio', 'cantidad', 'fecha_venta', 'categoria']
dtypes:
id_venta         int64
producto        object
precio          object
cantidad       float64
fecha_venta     object
categoria       object
dtype: object

Nulos por columna:
id_venta       0
producto       0
precio         2
cantidad       2
fecha_venta    0
categoria      0
dtype: int64

Primeras filas:


id_venta           producto   precio  cantidad fecha_venta  \
0       792  Cuadro decorativo   $69.94       5.0  02/01/2024   
1       811    Lámpara de mesa  $105.10       5.0  02/01/2024   
2      1156           Secadora   $97.96       3.0  02/01/2024   
3      1372           Heladera  $114.35       8.0  02/01/2024   
4      1546           Secadora  $106.21       4.0  02/01/2024   

           categoria  
0         Decoración  
1         Decoración  
2  Electrodomésticos  
3  Electrodomésticos  
4  Electrodomésticos


Análisis Descriptivo (numérico):


id_venta     cantidad
count  3035.000000  3033.000000
mean   1499.851400     6.496538
std     866.465379     3.457250
min       1.000000     1.000000
25%     748.500000     3.000000
50%    1502.000000     7.000000
75%    2249.500000     9.000000
max    3000.000000    12.000000

----------------------------------------------------------------------------------------------------


In [7]:
eda(clientes, "CLIENTES (Análisis Exploratorio)")

=== CLIENTES (Análisis Exploratorio) ===
shape: (567, 5)
columnas: ['id_cliente', 'nombre', 'edad', 'ciudad', 'ingresos']
dtypes:
id_cliente      int64
nombre         object
edad            int64
ciudad         object
ingresos      float64
dtype: object

Nulos por columna:
id_cliente    0
nombre        0
edad          0
ciudad        0
ingresos      0
dtype: int64

Primeras filas:


id_cliente               nombre  edad         ciudad  ingresos
0           1      Aloysia Screase    44  Mar del Plata  42294.68
1           2  Kristina Scaplehorn    25        Posadas  24735.04
2           3       Filip Castagne    50    Resistencia  35744.85
3           4          Liuka Luard    39   Bahía Blanca  27647.96
4           5        Dore Cockshtt    28        Rosario  28245.65


Análisis Descriptivo (numérico):


id_cliente        edad      ingresos
count  567.000000  567.000000    567.000000
mean   284.000000   37.940035  34668.739012
std    163.823075   10.202885  12974.531446
min      1.000000   20.000000    170.290000
25%    142.500000   30.000000  26015.240000
50%    284.000000   37.000000  35066.830000
75%    425.500000   43.000000  42457.100000
max    567.000000   81.000000  88053.010000

----------------------------------------------------------------------------------------------------


In [8]:
eda(marketing, "MARKETING (Análisis Exploratorio)")

=== MARKETING (Análisis Exploratorio) ===
shape: (90, 6)
columnas: ['id_campanha', 'producto', 'canal', 'costo', 'fecha_inicio', 'fecha_fin']
dtypes:
id_campanha       int64
producto         object
canal            object
costo           float64
fecha_inicio     object
fecha_fin        object
dtype: object

Nulos por columna:
id_campanha     0
producto        0
canal           0
costo           0
fecha_inicio    0
fecha_fin       0
dtype: int64

Primeras filas:


id_campanha         producto  canal  costo fecha_inicio   fecha_fin
0           74  Adorno de pared     TV   4.81   20/03/2024  03/05/2024
1           12           Tablet   RRSS   3.40   26/03/2024  13/05/2024
2           32  Lámpara de mesa  Email   5.54   28/03/2024  20/04/2024
3           21       Smartphone   RRSS   6.37   29/03/2024  16/05/2024
4           58         Alfombra  Email   4.25   31/03/2024  05/05/2024


Análisis Descriptivo (numérico):


id_campanha      costo
count    90.000000  90.000000
mean     45.500000   4.928667
std      26.124701   0.947750
min       1.000000   2.950000
25%      23.250000   4.372500
50%      45.500000   4.900000
75%      67.750000   5.562500
max      90.000000   7.390000

----------------------------------------------------------------------------------------------------


### Datasets cargados:
- Ventas: 1200 filas x 8 columnas
- Clientes: 500 filas x 6 columnas
- Marketing: 200 filas x 5 columnas

### Observaciones iniciales:
- Se detectaron 2 filas con valores nulos en el dataset de Ventas en la variables de `precio` y `cantidad`.


### 5️⃣ Calidad de Datos

In [9]:
#Función Control de Calidad de Datos


def calidad(df, nombre, clave=None):

    ## Mostrar título descriptivo con el nombre del DF
    print(f"=== {nombre} ===")


    ## Mostrar cantidad de valores nulos por columna
    print("Valores nulos por columna:")
    display(df.isna().sum().to_frame("nulos"))

    ## Contar filas duplicadas completas
    dup_rows = df.duplicated(keep=False).sum()
    print("Filas duplicadas (exactas):", dup_rows)

    if clave and clave in df.columns:
        # Contar cuántas filas tienen valores repetidos en esa columna
        dup_key = df[clave].duplicated(keep=False).sum()
        print(f"Duplicados por clave '{clave}':", dup_key)

        # Si existen duplicados, mostrar cuáles son los valores más repetidos
        if dup_key > 0:
            # Filtrar filas donde esa clave esté duplicado
            duplicados_ordenados = (
                df[df[clave].duplicated(keep=False)][clave]
                .value_counts()
                .sort_values(ascending=False)
            )

            print("\n🔁 Top valores duplicados más frecuentes:")
            # Mostrar solo los primeros 10 (los más repetidos)
            display(duplicados_ordenados.head(10))
        else:
            print(f"No se encontraron duplicados en la clave '{clave}'.")
    else:
        # Si la clave no fue pasada o no existe en el DataFrame
        if clave:
            print(f"La clave '{clave}' no existe en el DataFrame.")
        else:
            print("No se indicó una clave para analizar duplicados por columna.")

In [10]:
## Evaluar calidad de datos del dataset de Ventas
calidad(ventas, "VENTAS", clave="id_venta")

=== VENTAS ===
Valores nulos por columna:


nulos
id_venta         0
producto         0
precio           2
cantidad         2
fecha_venta      0
categoria        0

Filas duplicadas (exactas): 70
Duplicados por clave 'id_venta': 70

🔁 Top valores duplicados más frecuentes:


id_venta
56      2
421     2
424     2
1868    2
2545    2
2778    2
145     2
300     2
439     2
906     2
Name: count, dtype: int64

In [11]:
## Evaluar calidad de datos del dataset de Clientes
calidad(clientes, "CLIENTES", clave="id_cliente")

=== CLIENTES ===
Valores nulos por columna:


nulos
id_cliente      0
nombre          0
edad            0
ciudad          0
ingresos        0

Filas duplicadas (exactas): 0
Duplicados por clave 'id_cliente': 0
No se encontraron duplicados en la clave 'id_cliente'.


In [12]:
## Evaluar calidad de datos del dataset de Marketing
calidad(marketing, "MARKETING", clave="id_campanha")

=== MARKETING ===
Valores nulos por columna:


nulos
id_campanha       0
producto          0
canal             0
costo             0
fecha_inicio      0
fecha_fin         0

Filas duplicadas (exactas): 0
Duplicados por clave 'id_campanha': 0
No se encontraron duplicados en la clave 'id_campanha'.


## 🧹 Etapa 2: Preprocesamiento y Limpieza de Datos
**Objetivo:** Demostrar conocimiento de las técnicas de limpieza y transformación de datos.

### 1️⃣ Limpieza de datos

In [13]:
# Crear copias de dataframes
ventas_clean = ventas.copy()
clientes_clean = clientes.copy()
marketing_clean = marketing.copy()

# Eliminar filas completamente duplicadas
ventas_clean = ventas_clean.drop_duplicates()
clientes_clean = clientes_clean.drop_duplicates()
marketing_clean = marketing_clean.drop_duplicates()

In [14]:
calidad(ventas_clean, "VENTAS CLEAN", clave="id_venta")

=== VENTAS CLEAN ===
Valores nulos por columna:


nulos
id_venta         0
producto         0
precio           2
cantidad         2
fecha_venta      0
categoria        0

Filas duplicadas (exactas): 0
Duplicados por clave 'id_venta': 0
No se encontraron duplicados en la clave 'id_venta'.


In [15]:
# Función para limpiar texto en columnas tipo string
def normalizar_texto(df):
    for col in df.select_dtypes(include="object").columns:
        df[col] = (
            df[col]
            .astype(str)
            .str.strip()
            .str.replace(r"[\u200b\t\r\n]", "", regex=True)
            .str.replace(" +", " ", regex=True)
            .str.title()
        )
    return df

In [16]:
# Normalizar fechas
for df in [ventas_clean, clientes_clean, marketing_clean]:
    for col in df.columns:
        if "fecha" in col.lower():
            df[col] = pd.to_datetime(df[col], errors="coerce", dayfirst=True)

In [17]:
# Normalizar fechas de df Marketing

marketing_clean["fecha_inicio"] = pd.to_datetime(marketing_clean["fecha_inicio"], errors="coerce", dayfirst=True)
marketing_clean["fecha_fin"] = pd.to_datetime(marketing_clean["fecha_fin"], errors="coerce", dayfirst=True)

In [18]:
# Verificar el tipo de dato
print(ventas_clean.dtypes)
print()
print(clientes_clean.dtypes)
print()
print(marketing_clean.dtypes)
print()

id_venta                int64
producto               object
precio                 object
cantidad              float64
fecha_venta    datetime64[ns]
categoria              object
dtype: object

id_cliente      int64
nombre         object
edad            int64
ciudad         object
ingresos      float64
dtype: object

id_campanha              int64
producto                object
canal                   object
costo                  float64
fecha_inicio    datetime64[ns]
fecha_fin       datetime64[ns]
dtype: object



In [19]:
#  Aplicar la normalización de texto
ventas_clean = normalizar_texto(ventas_clean)
clientes_clean = normalizar_texto(clientes_clean)
marketing_clean = normalizar_texto(marketing_clean)

In [20]:
# Mostrar los df luego de normalizar los textos para revisar que queden bien
print(ventas_clean.head(10))
print(clientes_clean.head(10))
print(marketing_clean.head(10))

   id_venta           producto   precio  cantidad fecha_venta  \
0       792  Cuadro Decorativo   $69.94       5.0  2024-01-02   
1       811    Lámpara De Mesa  $105.10       5.0  2024-01-02   
2      1156           Secadora   $97.96       3.0  2024-01-02   
3      1372           Heladera  $114.35       8.0  2024-01-02   
4      1546           Secadora  $106.21       4.0  2024-01-02   
5      1697    Horno Eléctrico   $35.35       9.0  2024-01-02   
6      1710   Plancha De Vapor   $65.43       2.0  2024-01-02   
7      2959          Proyector   $88.17       9.0  2024-01-02   
8       318  Rincón De Plantas   $79.86      11.0  2024-01-03   
9       419         Candelabro   $66.11       8.0  2024-01-03   

           categoria  
0         Decoración  
1         Decoración  
2  Electrodomésticos  
3  Electrodomésticos  
4  Electrodomésticos  
5  Electrodomésticos  
6  Electrodomésticos  
7        Electrónica  
8         Decoración  
9         Decoración  
   id_cliente               nom

In [21]:
# Normalizar valores numéricos

if "precio" in ventas_clean.columns:
    ventas_clean["precio"] = (
        ventas_clean["precio"]
        .astype(str)
        .str.replace("$", "", regex=False)
        .str.replace(",", "", regex=False)
        .str.strip()
    )
    ventas_clean["precio"] = pd.to_numeric(ventas_clean["precio"], errors="coerce")

In [22]:
# Verificar el df ventas_clean

print(ventas_clean.dtypes)
print()
print(ventas_clean.columns)

id_venta                int64
producto               object
precio                float64
cantidad              float64
fecha_venta    datetime64[ns]
categoria              object
dtype: object

Index(['id_venta', 'producto', 'precio', 'cantidad', 'fecha_venta',
       'categoria'],
      dtype='object')


In [23]:
# Verificar si hay valores nulos en "cantidad"

if "cantidad" in ventas_clean.columns:
    ventas_clean["cantidad"] = pd.to_numeric(
        ventas_clean["cantidad"], errors="coerce"
    ).astype("Int64")


In [24]:
print(ventas_clean.dtypes)

id_venta                int64
producto               object
precio                float64
cantidad                Int64
fecha_venta    datetime64[ns]
categoria              object
dtype: object


In [25]:
# Guardar los DataFrames limpios como CSV

ventas_clean.info()
ventas_clean.to_csv("/content/drive/MyDrive/Data Analytics con Python/Perea Daniela - Comisión 25262 - TPI Data Analytics - Pre-entrega/Datasets/ventas_clean.csv", index=False)
clientes_clean.to_csv("/content/drive/MyDrive/Data Analytics con Python/Perea Daniela - Comisión 25262 - TPI Data Analytics - Pre-entrega/Datasets/clientes_clean.csv", index=False)
marketing_clean.to_csv("/content/drive/MyDrive/Data Analytics con Python/Perea Daniela - Comisión 25262 - TPI Data Analytics - Pre-entrega/Datasets/marketing_clean.csv", index=False)

print("✅ Archivos guardados: ventas_clean.csv, clientes_clean.csv, marketing_clean.csv")

<class 'pandas.core.frame.DataFrame'>
Index: 3000 entries, 0 to 3034
Data columns (total 6 columns):
 #   Column       Non-Null Count  Dtype         
---  ------       --------------  -----         
 0   id_venta     3000 non-null   int64         
 1   producto     3000 non-null   object        
 2   precio       2998 non-null   float64       
 3   cantidad     2998 non-null   Int64         
 4   fecha_venta  3000 non-null   datetime64[ns]
 5   categoria    3000 non-null   object        
dtypes: Int64(1), datetime64[ns](1), float64(1), int64(1), object(2)
memory usage: 167.0+ KB
✅ Archivos guardados: ventas_clean.csv, clientes_clean.csv, marketing_clean.csv


In [26]:
print(ventas_clean.select_dtypes(include="object").columns)

Index(['producto', 'categoria'], dtype='object')


In [27]:
# Reporte Global de Calidad de Datos (de los 3 df)

def reporte_calidad_global(dfs, nombres):
    resumen = []
    for df, nombre in zip(dfs, nombres):
        nulos = df.isna().sum().sum()
        duplicados = df.duplicated(keep=False).sum()
        columnas = len(df.columns)
        filas = len(df)

        resumen.append({
            "Dataset": nombre,
            "Filas": filas,
            "Columnas": columnas,
            "Nulos totales": nulos,
            "Duplicados": duplicados,
        })

    reporte = pd.DataFrame(resumen)
    return reporte


In [28]:
# Presentación del reporte

print(reporte_calidad_global([ventas, clientes, marketing], ["VENTAS Original", "CLIENTES Original", "MARKETING Original"]))
print(reporte_calidad_global([ventas_clean, clientes_clean, marketing_clean],["VENTAS Copia   ", "CLIENTES Copia   ", "MARKETING Copia   "]))

              Dataset  Filas  Columnas  Nulos totales  Duplicados
0     VENTAS Original   3035         6              4          70
1   CLIENTES Original    567         5              0           0
2  MARKETING Original     90         6              0           0
              Dataset  Filas  Columnas  Nulos totales  Duplicados
0     VENTAS Copia      3000         6              4           0
1   CLIENTES Copia       567         5              0           0
2  MARKETING Copia        90         6              0           0


### 2️⃣ Transformación de datos

In [29]:
# Funcion para encontrar columnas con nombres

def encontrar_columna(df, candidatos):
    for c in df.columns:
        nombre = c.lower()
        if any(p in nombre for p in candidatos):
            return c
    return None

In [30]:
# Detectar la columna de producto

prod_col = encontrar_columna(ventas_clean, ["producto", "id_producto", "sku", "articulo", "artículo"])
if prod_col is None:
    raise ValueError("No se encontró columna de producto. Renombrá una columna a 'producto' o similar.")
print(prod_col)


producto


In [31]:
# Calcular ingreso por registro = precio * cantidad

ventas_perf = (
    ventas_clean
    .assign(
        ingreso = ventas_clean["precio"] * ventas_clean["cantidad"]
    )
)

In [32]:
# Agregar métricas por producto

resumen_prod = (
    ventas_perf
    # 1) Agrupar el DataFrame por una o varias columnas clave
    .groupby(
        by=prod_col,
        dropna=False,
        as_index=False,
        observed=False
    )
    # 2) Agregamos (resumimos) columnas numéricas por cada grupo
    .agg(
        ingreso_total=('ingreso', 'sum'),
        unidades=('cantidad', 'sum'),
        precio_promedio=('precio', 'mean'),
        registros=('ingreso', 'size')
    )
)

In [33]:
# Verificar tabla de ventas

print(resumen_prod.head(10))

                 producto  ingreso_total  unidades  precio_promedio  registros
0         Adorno De Pared       48093.49       633        76.097800        100
1                Alfombra       44773.06       615        74.098300        100
2              Aspiradora       50085.86       651        77.447400        100
3             Auriculares       74175.58       958        76.302727        143
4                Batidora        50979.2       672        77.537200        100
5                Cafetera       59607.31       765        79.046581        117
6              Candelabro        11128.8       160        74.595417         24
7  Consola De Videojuegos       46174.41       623        76.346869         99
8                Cortinas       44865.03       610        73.649900        100
9       Cuadro Decorativo        54297.6       726        74.578000        100


In [34]:
# Calcular percentil 80 de ingreso_total

p80_ingreso = resumen_prod["ingreso_total"].quantile(
    q=0.80,
    interpolation="linear"
)

In [35]:
# Filtrar los productos "de alto rendimiento" y ordenarlos

## Filtrar
ventas_top = (
    resumen_prod
    .query(
        "ingreso_total >= @p80_ingreso",
        engine="python"
    )

  ## Ordenar
    .sort_values(
        by=["ingreso_total", "unidades"],
        ascending=[False, False],
        na_position="last",
        ignore_index=True
    )
)

# Mostrar resultados
print(f"Columna de producto detectada: {prod_col}")
print(f"Umbral (percentil 80) de ingreso_total: {float(p80_ingreso):,.2f}")
print("✅ Productos de ALTO RENDIMIENTO (top 20% por ingreso):")
display(ventas_top.head(20))

Columna de producto detectada: producto
Umbral (percentil 80) de ingreso_total: 52,518.85
✅ Productos de ALTO RENDIMIENTO (top 20% por ingreso):


producto  ingreso_total  unidades  precio_promedio  registros
0    Lámpara De Mesa       82276.38      1112        72.720625        176
1        Auriculares       74175.58       958        76.302727        143
2         Microondas       72562.89       912        79.176000        135
3           Cafetera       59607.31       765        79.046581        117
4  Cuadro Decorativo        54297.6       726        74.578000        100
5         Smartphone       54132.44       665        81.398416        101

### 💡 Interpretación

- Los **6 productos** listados representan el **20% superior** de la cartera, concentrando una porción significativa de los ingresos totales.  
- Se destacan la **Lámpara de Mesa** y los **Auriculares**, con ingresos superiores a $70.000 y volúmenes de venta altos.

- La mayoría de los productos top presentan un precio promedio estable entre $ 70.000  y  $ 80.000, lo que sugiere una relación equilibrada entre precio y demanda.

📈 **Conclusión:** este análisis permite identificar los productos más rentables y definir estrategias focalizadas, como priorizar su stock, campañas de marketing o bundles de venta.

### 3️⃣ Agregación

In [36]:
# Funcion para encontrar la columna de categoría
def encontrar_columna(df, candidatos):

    for c in df.columns:
        if any(p in c.lower() for p in candidatos):
            return c
    return None

In [37]:
# Detectar columna de categoría (acepta variantes)
cat_col = encontrar_columna(ventas_clean, ["categoria", "categoría", "categoria_producto", "rubro"])
if cat_col is None:
    raise ValueError("No se encontró columna de categoría (por ej. 'categoria' o 'rubro').")

In [38]:
# Asegurar columna 'ingreso' (si no existe, crearla)
if "ingreso" not in ventas_clean.columns:
    ventas_cat = ventas_clean.assign(ingreso = ventas_clean["precio"] * ventas_clean["cantidad"])
else:
    ventas_cat = ventas_clean.copy()

In [39]:
# Agregación por categoría con groupby + agg
resumen_cat = (
    ventas_cat
    .groupby(
        by=cat_col,
        dropna=False,
        as_index=False
    )
    .agg(
        ingreso_total=('ingreso', 'sum'),
        unidades=('cantidad', 'sum'),
        ventas=('ingreso', 'size'),
        precio_promedio=('precio', 'mean')
    )
    .sort_values(
        by='ingreso_total',
        ascending=False,
        na_position='last',
        ignore_index=True
    )
)

In [40]:
# Ticket promedio por venta = ingreso_total / ventas
resumen_cat = resumen_cat.assign(
    ticket_promedio_por_venta = resumen_cat['ingreso_total'] / resumen_cat['ventas']
)

print("Columna de categoría detectada:", cat_col)
print("Resumen por categoría (ordenado por ingreso_total):")
display(resumen_cat.head(20))

Columna de categoría detectada: categoria
Resumen por categoría (ordenado por ingreso_total):


categoria  ingreso_total  unidades  ventas  precio_promedio  \
0  Electrodomésticos      505299.63      6592    1000         76.52096   
1        Electrónica       482577.8      6413     999         75.25492   
2         Decoración      479216.09      6490    1001         74.09800   

   ticket_promedio_por_venta  
0                  505.29963  
1                 483.060861  
2                 478.737353

### 📊 *Insights principales*

- 🧺 **Electrodomésticos** lidera en **ingresos totales (505.000)**, combinando un volumen medio con el ticket promedio más alto.  
- 🪞 **Decoración** presenta la **mayor cantidad de ventas (1,001)**, pero con el **ticket promedio más bajo**, lo que sugiere compras más frecuentes de bajo valor.  
- 💻 **Electrónica** mantiene valores intermedios en todos los indicadores, mostrando un **equilibrio entre volumen y rentabilidad**.  
- 💲 Los **precios promedios son similares (~$75.000)** entre categorías, lo que refleja una política de precios coherente.  

### 💡 *Conclusión*

**Electrodomésticos** es la categoría **más rentable** al combinar un **ticket promedio alto** con un **volumen sólido de ventas**, mientras que **Decoración** se destaca por la **frecuencia de compra** y **Electrónica** mantiene un **rendimiento estable y competitivo**.  
